### PR TITLE
Update queries with lack of coverage in checking version and flags for Docker Closes #1995

### LINKED ISSUE #1995: Update queries with lack of coverage in checking version and flags for Docker
### Platform
Docker

### Description
Some queries return false positives since they do not take into account flags between the command and various types of version

### PR DESCRIPTION
**Pull request by rafaela-soares**
Closes #1995

**Proposed Changes**

- **Queries "Zypper Install Without Version" (562952e4-0348-4dea-9826-44f3a2c6117b)**, **"Yum install Without Version" (6452c424-1d92-4deb-bb18-a03e95d579c4)**, **"Unpinned Package Version in Pip Install" (02d9c71f-3ee8-4986-9c27-1a20d0d19bfc)**, **"Gem Install Without Version" (22cd11f7-9c6c-4f6e-84c0-02058120b341)**: This queries should consider flags between the commands and various types of version. Some of them just considered "http-2.2.2" or "http=2.2.2". However, they have to consider both and "http=$VAR_HTTP_VERSION"
- **Query "Yum Install Allows Manual Input" (6e19193a-8753-436d-8a09-76dcff91bb03)**, **"Apt Get Install Pin Version Not Defined" (965a08d7-ef86-4f14-8792-4a3b2098937e)**, **"Apt Get Install Lists Were Not Deleted" (df746b39-6564-4fed-bf85-e9c44382303c)**: This query should consider flags between the commands

### CODE REVIEW

**Comment by rafaela-soares** [comment]
I have to redo. Please, do not review this branch. Sorry

File: assets/queries/dockerfile/pin_versions_apt_get_install/query.rego [hunk 1]
```diff
@@ -1,31 +1,54 @@
...
+		"keyExpectedValue": sprintf("Package '%s' has version defined", [resource.Value[j]]),
+		"keyActualValue": sprintf("Package '%s' does not have version defined", [resource.Value[j]]),
 	}
 }
 
-noVersion(pack) {
-	not contains(pack, "=")
+isAptGet(command) {
+	contains(command[x], "apt-get")
+	contains(command[j], "install")
```
Comment (joaomartinscx) [inline assets/queries/dockerfile/pin_versions_apt_get_install/query.rego:53]: Perhaps its a good idea to establish a relation between *x* and *j*

**Review by joaomartinscx — CHANGES_REQUESTED** [review]

File: assets/queries/dockerfile/gem_install_without_version/query.rego [hunk 2]
```diff
@@ -1,10 +1,43 @@
...
-	not contains(resource.Original, ":")
+
+	count(resource.Value) > 1
+
+	isGem(resource.Value)
+
+	resource.Value[j] != "install"
+	resource.Value[j] != "gem"
+	regex.match("^[a-zA-Z]", resource.Value[j]) == true
+	not dockerLib.withVersion(resource.Value[j])
```
[the remark this answers is not in the document]
Comment (rafaela-soares) [reply assets/queries/dockerfile/gem_install_without_version/query.rego:40]: No. That's why the line 39 exists. The regex regex.match("^[a-zA-Z]", resource.Value[j]) == true just catches words beginning with letters

File: assets/queries/dockerfile/pin_versions_apt_get_install/query.rego (hunk 1 above)
Comment (rafaela-soares) [reply assets/queries/dockerfile/pin_versions_apt_get_install/query.rego:53]: It is to verify if different indexes have "apt-get" and "install".
For example, RUN ["apt-get", "install", "blunder"]

File: assets/queries/dockerfile/gem_install_without_version/query.rego [hunk 3]
```diff
@@ -14,3 +47,8 @@ CxPolicy[result] {
 		"keyActualValue": sprintf("%s is 'gem install <gem>', you should use 'gem install <gem>:<version>", [resource.Original]),
 	}
 }
+
+isGem(command) {
+	contains(command[x], "gem")
+	contains(command[j], "install")
```
[the remark this answers is not in the document]
Comment (rafaela-soares) [reply assets/queries/dockerfile/gem_install_without_version/query.rego:53]: It is to verify if different indexes have "gem" and "install".
For example, RUN ["gem", "install", "blunder"]

File: assets/queries/dockerfile/yum_install_without_version/query.rego [hunk 4]
```diff
@@ -1,29 +1,54 @@
...
 
 	result := {
-		"documentId": document.id,
-		"searchKey": sprintf("FROM={{%s}}.{{%s}}", [img, commands[img][c].Original]),
+		"documentId": input.document[i].id,
+		"searchKey": sprintf("FROM={{%s}}.{{%s}}", [name, commands]),
 		"issueType": "IncorrectValue",
-		"keyExpectedValue": "The package version should always be specified when using yum install",
-		"keyActualValue": sprintf("No version is specified in '%s'", [commands[img][c].Original]),
+		"keyExpectedValue": "The package version should always be specified when using zypper install",
```
Comment (ruigomescx) [inline assets/queries/dockerfile/yum_install_without_version/query.rego:24]: did meant zypper or yum?

File: assets/queries/dockerfile/yum_install_without_version/query.rego [hunk 5]
```diff
@@ -1,29 +1,54 @@
...
+	resource.Value[j] != "install"
+	resource.Value[j] != "yum"
+	regex.match("^[a-zA-Z]", resource.Value[j]) == true
+	not dockerLib.withVersion(resource.Value[j])
+
+	result := {
+		"documentId": input.document[i].id,
+		"searchKey": sprintf("FROM={{%s}}.{{%s}}", [name, resource.Original]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": "The package version should always be specified when using zypper install",
```
Comment (ruigomescx) [inline assets/queries/dockerfile/yum_install_without_version/query.rego:46]: see previous comment

**Review by ruigomescx — CHANGES_REQUESTED** [review]

File: assets/queries/dockerfile/yum_install_without_version/query.rego (hunk 4 above)
Comment (rafaela-soares) [reply assets/queries/dockerfile/yum_install_without_version/query.rego:24]: Yum, sorry

**Review by ruigomescx — APPROVED** [review]

**Pull request merged**